test/lib/ansible_test/_util/target/sanity/import/importer.py

```diff
@@ -149,10 +157,32 @@ def __init__(self, path, name, restrict_to_module_paths):
             self.loaded_modules = set()
             self.restrict_to_module_paths = restrict_to_module_paths
 
+        def find_spec(self, fullname, path=None, target=None):  # pylint: disable=unused-argument
+            """Return the spec from the loader or None
+            :param fullname: str
+            :param path: List[str]
+            :param target: str
+            :return: ModuleSpec | None
```

Comment:
The `:param` and `:return:` docstring entries here (and below) should be replaced comment based type hints.

The existing ones are also incorrect and should be type comments instead.

Reply:
There are also quite a few `:type` type comments that were never converted to PEP 484 syntax. While those should also be fixed, I can take care of that in a separate PR once this is merged.

Reply:
Note that `:param:` entries should contain text descriptions, not types. Same for `:returns:`. There's a separate `:rtype:` for the type but with type annotation comments it won't be necessary.

Reply:
Fixed, and rebased this to test the type hints in CI too.

Reply:
Looks like CI found broken stuff though, so I'll need some help figuring out how to test these properly.

Reply:
Hrm, that's weird- we've got blanket `ignore_missing_imports` for both `importlib.*` and `collections.*`, which should prevent the error you're getting. That's documented to also apply to subpackages, but maybe try adding `[mypy-collection.abc.*]` et al to that file in your PR to see if that fixes it?

https://github.com/ansible/ansible/commit/3d5637beecbbdcd1bc11a85d180de2087ecf12ee#diff-ed32086a44539250b26aa1f15420c7d79aad51f971a3126aece5ac419b3efdb5R91-R95

Reply:
That did fix it. The issue was caused by running the mypy test for this file against remote-only python versions (in addition to controller python versions). I simplified things in the last commit because I'm not sure whether or not it's helpful (the other mypy test targets in test/lib/ansible_test/_internal/ are only tested against controller python versions), but I can drop that commit removing that if it seems useful.

Reply:
It's not a big deal either way, but since import sanity actually has to run on 2.x, it'd be *nice* if the type hint support was getting exercised there as well. Honestly I don't have strong feelings either way- I'm happy to merge as-is and we can go fix it and turn on remote-side stuff later (or not), or we can wait for Matt C to be back late next week and see what he says.